### PR TITLE
feat: adding build executable to worker

### PR DESCRIPTION
**Pull request by claire1618**

### CODE REVIEW

File: cmd/vela-worker/exec.go [hunk 1]
```diff
@@ -60,6 +62,17 @@ func (w *Worker) exec(index int, config *library.Worker) error {
 		return err
 	}
 
+	execBuildExecutable, resp, err := execBuildClient.Build.GetBuildExecutable(item.Repo.GetOrg(), item.Repo.GetName(), item.Build.GetNumber())
```
Comment (ecrupper) [inline cmd/vela-worker/exec.go:65]: Can we add a comment above this call to explain what is going on?

File: cmd/vela-worker/exec.go [hunk 2]
```diff
@@ -60,6 +62,17 @@ func (w *Worker) exec(index int, config *library.Worker) error {
...
+	if err != nil {
+		return err
+	}
+
+	var execBuildExecutablePipeline pipeline.Build
+	err = json.Unmarshal([]byte(execBuildExecutable.GetData()), &execBuildExecutablePipeline)
```
Comment (ecrupper) [inline cmd/vela-worker/exec.go:71]:
```suggestion
	pipeline := new(pipeline.Build)
	err = json.Unmarshal(execBuildExecutable.GetData()), pipeline)
```

File: cmd/vela-worker/exec.go [hunk 3]
```diff
@@ -149,7 +162,7 @@ func (w *Worker) exec(index int, config *library.Worker) error {
 		Hostname:            w.Config.API.Address.Hostname(),
 		Runtime:             w.Runtime,
 		Build:               item.Build,
-		Pipeline:            item.Pipeline.Sanitize(w.Config.Runtime.Driver),
+		Pipeline:            &execBuildExecutablePipeline,
```
Comment (ecrupper) [inline cmd/vela-worker/exec.go:165]:
```suggestion
		Pipeline:            pipeline.Sanitize(w.Config.Runtime.Driver),
```
we still want that sanitize function there 👍

File: executor/linux/build.go [hunk 4]
```diff
@@ -369,6 +369,7 @@ func (c *client) AssembleBuild(ctx context.Context) error {
 
 		// assume no privileged images are in use
 		containsPrivilegedImages := false
+		var privImage string
```
Comment (ecrupper) [inline executor/linux/build.go:372]: These privileged image check changes look like they are from a different story? I say open a separate PR for this.

**Review by ecrupper — CHANGES_REQUESTED** [review]

File: cmd/vela-worker/exec.go (hunk 1 above)
Comment (claire1618) [reply cmd/vela-worker/exec.go:66]: Yes!

File: cmd/vela-worker/exec.go (hunk 2 above)
Comment (claire1618) [reply cmd/vela-worker/exec.go]: I updated the code with those changes

File: cmd/vela-worker/exec.go (hunk 3 above)
Comment (claire1618) [reply cmd/vela-worker/exec.go]: I updated the code to reflect that

File: executor/linux/build.go (hunk 4 above)
Comment (claire1618) [reply executor/linux/build.go]: will do!

File: cmd/vela-worker/exec.go [hunk 5]
```diff
@@ -60,6 +62,19 @@ func (w *Worker) exec(index int, config *library.Worker) error {
 		return err
 	}
 
+	// request build executable containing pipeline.Build data using exec client
+	execBuildExecutable, resp, err := execBuildClient.Build.GetBuildExecutable(item.Repo.GetOrg(), item.Repo.GetName(), item.Build.GetNumber())
```
Comment (plyr4) [inline cmd/vela-worker/exec.go:66]: following the golangci failure, if we dont need `resp` can we omit the variable? 
```suggestion
	execBuildExecutable, _, err := execBuildClient.Build.GetBuildExecutable(item.Repo.GetOrg(), item.Repo.GetName(), item.Build.GetNumber())
```

Comment (claire1618) [reply cmd/vela-worker/exec.go]: I updated the code with that change

**Review by ecrupper — APPROVED** [review]

**Review by ecrupper — APPROVED** [review]

**Pull request merged**